libs/commons/Common.ml

```diff
@@ -828,21 +828,27 @@ let cat file =
    unclear.
 *)
 let read_file ?(max_len = max_int) path =
-  let buf_len = 4096 in
-  let extbuf = Buffer.create 4096 in
-  let buf = Bytes.create buf_len in
-  let rec loop fd =
-    match Unix.read fd buf 0 buf_len with
-    | 0 -> Buffer.contents extbuf
-    | num_bytes ->
-        assert (num_bytes > 0);
-        assert (num_bytes <= buf_len);
-        Buffer.add_subbytes extbuf buf 0 num_bytes;
-        if Buffer.length extbuf >= max_len then Buffer.sub extbuf 0 max_len
-        else loop fd
-  in
-  let fd = Unix.openfile path [ Unix.O_RDONLY ] 0 in
-  Fun.protect ~finally:(fun () -> Unix.close fd) (fun () -> loop fd)
+  if !jsoo then (
+    let ic = open_in path in
+    let s = really_input_string ic (in_channel_length ic) in
+    close_in ic;
+    s)
```

Comment:
We're going to need to make a similar change to https://github.com/returntocorp/ocaml-tree-sitter-core/blob/main/src/run/lib/Util_file.ml#L25-L41